### DISCUSSION TITLE
ntptime, how to determine offset hours

### DISCUSSION THREAD

**Discussion by davefes** [micropython/micropython#11270] [Using MicroPython] [answered]
Using this version:
https://github.com/peterhinch/micropython-samples/blob/master/ntptime/ntptime.py
and wondered how to determine the offset, especially when it changes throughout the year.

Is there someway to determine offset hours without a previous call to ntptime()?

**Comment by davefes** [comment]
OK, I see in this implementation:
https://gist.github.com/Beormund/f0f39c72e066da497f6308d1964c9627
offset_hours being past into time().

**Comment by obdevel** [comment, marked as answer]
A more complex method is to use google geocode and timezone APIs. The former will resolve a location to a lat/long pair from which the latter can produce the timezone, and the current UTC and DST offsets. Needs HTTPS client and JSON parsing modules.

**Reply by davefes** [reply]
Thanks.  Just finished some testing with  https://gist.github.com/Beormund/f0f39c72e066da497f6308d1964c9627
and that is good enough for my needs.
